### ISSUE TITLE
why does jedis about class JedisCluster and JedisPool did not implement some common interface for overwrite the same method

### ISSUE THREAD

**Issue by gwdgithubnom** [redis/jedis#2585]
### Expected behavior
In our development, we expected to use jedis in different env, some time the redis is cluster mode, but some time is not cluster mode. such as, we did the test env is not cluster mode redis. So we need to write an adapt code to make JedisCluster and JedisPool to be compatible. We think this is very common demand for the jedis. But, didn't understand the design of the JedisCluster and JedisPool with no common interface

### Actual behavior
none

### Steps to reproduce:
none

### Redis / Jedis Configuration
none

#### Jedis version:
3.5+
#### Redis version:
3.1+
#### Java version:
1.8

**Comment by yangbodong22011** [comment]
If I understand correctly, in fact, you need a `getConnection` method, just return Jedis.

```
  /**
   * Get a connection from the connection pool, and finally call jedis.close to
   * return it back to the connection pool after the request is over.
   * @param key the key
   * @return Jedis connection
   */
  private Jedis getConnection(final String key) {
    Jedis jedis = null;
    if (clusterEnabled) {
      jedis = jedisCluster.getConnectionFromSlot(JedisClusterCRC16.getSlot(key));
    } else {
      jedis = jedisPool.getResource();
    }
    return jedis;
  }
```

then, in your code : 
```
try (Jedis jedis = getConnection(key)) {
  if (jedis.hmset(key, StringByteIterator.getStringMap(values))
      .equals("OK")) {
    jedis.zadd(INDEX_KEY, hash(key), key);
    return Status.OK;
  }
} catch (Exception e) {
  return Status.ERROR;
}
```

you can refer this: https://github.com/brianfrankcooper/YCSB/pull/1535/files, don’t know if it’s what you want.

**Comment by gwdgithubnom** [comment]
> If I understand correctly, in fact, you need a `getConnection` method, just return Jedis.
> 
> ```
>   /**
>    * Get a connection from the connection pool, and finally call jedis.close to
>    * return it back to the connection pool after the request is over.
>    * @param key the key
>    * @return Jedis connection
>    */
>   private Jedis getConnection(final String key) {
>     Jedis jedis = null;
>     if (clusterEnabled) {
>       jedis = jedisCluster.getConnectionFromSlot(JedisClusterCRC16.getSlot(key));
>     } else {
>       jedis = jedisPool.getResource();
>     }
>     return jedis;
>   }
> ```
> 
> then, in your code :
> 
> ```
> try (Jedis jedis = getConnection(key)) {
>   if (jedis.hmset(key, StringByteIterator.getStringMap(values))
>       .equals("OK")) {
>     jedis.zadd(INDEX_KEY, hash(key), key);
>     return Status.OK;
>   }
> } catch (Exception e) {
>   return Status.ERROR;
> }
> ```
> 
> you can refer this: https://github.com/brianfrankcooper/YCSB/pull/1535/files, don’t know if it’s what you want.

thank you for reply， i did checked the code, found this issue was fixed in 2021/05/17, about #2533.
